src/append/rolling_file/policy/compound/trigger/time.rs

```diff
@@ -291,12 +304,11 @@ impl Trigger for TimeTrigger {
 
         #[cfg(not(test))]
         let current: DateTime<Local> = Local::now();
-        let mut next_roll_time = self.next_roll_time.write().unwrap();
+        let next_roll_time = self.next_roll_time.read().unwrap();
         let is_trigger = current >= *next_roll_time;
+        drop(next_roll_time);
         if is_trigger {
-            let tmp = TimeTrigger::new(self.config);
-            let time_new = tmp.next_roll_time.read().unwrap();
-            *next_roll_time = *time_new;
+            self.refresh_time();
         }
         Ok(is_trigger)
     }
```

Comment:
It won't let me add the comment below, but do we need to keep the `pub(crate)` below? On the Deserializer

Reply:
We don't use it on other triggers and the tests pass with out it. It looks to me like just `pub` will do without the full crate visibility